common/utils/src/main/resources/error/error-conditions.json

```diff
@@ -1914,6 +1914,44 @@
     ],
     "sqlState" : "22012"
   },
+  "INTERVAL_ERROR" : {
+    "message" : [
+      "Interval error."
+    ],
+    "subClass" : {
+      "DAY_TIME_PARSING" : {
```

Comment:
We already have this error condition `INVALID_INTERVAL_FORMAT`. Can't you reuse it?

Reply:
Yes, however, this error and its subclasses consider arithmetic intervals, and this one that I added considers date interval errors. If you recommend, I can rename the parent class to something like `DATE_INTERVAL_ERROR`

Reply:
Spark doesn't have an arithmetic intervals, only datetime intervals.

Let's think of moving some interval parsing errors to `INVALID_INTERVAL_FORMAT`.